### PR TITLE
ci: prefer io.write instead of print to have full control of output

### PR DESCRIPTION
**Pull request by dundargoc**
This should prevent different output formats on windows and unix.

### CODE REVIEW

**Comment by dundargoc** [comment]
https://github.com/nvim-treesitter/nvim-treesitter/pull/2421 needs to be merged first.

**Comment by dundargoc** [comment]
`print()` outputs an extra newline on windows jobs. I suspect Github Actions interprets carriage return as a newline as well.

**Comment by dundargoc** [comment]
Extra context: `print()` outputs an extra newline on windows jobs for some reason, making the output jarring to read.

File: scripts/check-queries.lua [hunk 1]
```diff
@@ -35,7 +35,7 @@ local function do_check()
   local captures = extract_captures()
   local last_error
 
-  print "::group::Check parsers"
+  io.write "::group::Check parsers\n"
```
Comment (theHamsta) [inline scripts/check-queries.lua:38]: maybe add a local wrapper around that idiom?

Comment (theHamsta) [reply scripts/check-queries.lua:38]: but would be also fine with this solution.

Comment (dundargoc) [reply scripts/check-queries.lua:38]: > maybe add a local wrapper around that idiom?

Sorry, I'm still new to lua. What do you mean? Can you give an example/link?

Comment (theHamsta) [reply scripts/check-queries.lua:38]: write your own `print` that uses `io.write`. Like `local function io_print(text) io.write(text .. '\n') end`

**Review by theHamsta — APPROVED** [review]

**Comment by theHamsta** [comment]
Thank you! I found those empty lines also very annoying for a long time but never found the motivation to search for the reason.

**Pull request merged**